src/context.js

```diff
@@ -1,10 +1,17 @@
 "use strict";
 module.exports = function(Promise) {
+
+var util = require("./util");
 var longStackTraces = false;
 var contextStack = [];
 
-Promise.prototype._promiseCreated = function() {};
-Promise.prototype._pushContext = function() {};
+util.hookTo(Promise.prototype, "_hook_created", null);
```

Comment:
There is no sense to turn monitoring feature on and off because it does not add self executed code, it only adds methods that could be invoked, impacting the library size only. It could be made separate optional module, like "call_get".
Performance measurement results (doxbee-sequential):

__before the change:__
promises-bluebird-generator.js                  250       29.55
promises-bluebird.js                            359       45.42
__after the change:__
promises-bluebird-generator.js                  228       29.32
promises-bluebird.js                            360       45.05

I see that the variance of the benchmark is much higher than between 2 measurements, so we can conclude that there is no performance degradation.